Developer experience: Add EchoSQLLogger

amenk opened the pull request:
### 1. Why is this change necessary?

It improves developer experience by providing a way to print queries, for example in tests, for further debugging.
There is already the EchoSQLLogger from Doctrine, but it is less useful in Shopware because

1. UUIDs are printed as binary parameters, rendering them unreadable / unusable
2. Parameters are not inlined as in the debug bar, making it harder to analyse them in the SQL tool of one's choice by copy&paste.

### 2. What does this change do, exactly?

It adds a SQL Logger which is similar to what the debug bar offers.

### 3. Describe each step to reproduce the issue or behaviour.

When debugging, add

     Kernel::getConnection()->getConfiguration()->setSQLLogger(
         new \Shopware\Core\Profiling\Doctrine\EchoSQLLogger()
     );

This will print out the SQL quries

### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [ ] I have written tests and verified that they fail without my change - not added because of trivial functionality
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes - if the MR gets accepted, I will add the procedure to shopware/docs
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

amenk committed:
Developer experience: Add EchoSQLLogger

leichteckig commented:
Hello @amenk! Thank you for your contribution 💙   Before we proceed, could you take a look at the code style issues mentioned [here](https://github.com/shopware/platform/pull/1880/checks?check_run_id=2646265862)?

amenk commented:
@leichteckig Thanks for the quick feedback, I fixed the code style issues now and rebased on `trunk`
PHPUnit was also failing, do you know if this was related to my pull request?

leichteckig commented:
@amenk The PHPUnit result is not caused by your changes. We're on it. 🙂

J-Rahe commented:
Hey @amenk, thanks for opening this PR : ). I've just discussed your PR and we came to the conclusion that this may prove useful to some, but should not go into the core. You could instead make this into an basic plugin for your dev-environment.
Sorry that we were not able to give feedback earlier.

amenk commented:
Yes, it would have been better to decline it before requesting Style Fixes .. anyways, thanks for discussing :-)
